include/aws/http/http2_stream_manager.h

```diff
@@ -51,9 +51,12 @@ struct aws_http2_stream_manager_options {
     /**
      * If TLS options is set, you also need to handle ALPN, otherwise, may not able to get HTTP/2 connection and fail
      * the stream manager.
-     * If TLS options not set, prior knowledge will be used.
+     * If TLS options not set, you can set prior_knowledge to true to establish HTTP/2 connection. We don't support
+     * upgrade from HTTP/1 connection.
      */
     const struct aws_tls_connection_options *tls_connection_options;
+    /* Use prior knowledge or not. Error out when TLS is set and this is true. */
+    const bool prior_knowledge;
```

Comment:
trivial: `const bool` isn't common in our codebase, we generally just use `const` for pointers, but don't bother marking by-value variables with `const`